src/network/netlink.rs

```diff
@@ -72,48 +218,36 @@ impl std::fmt::Display for Route {
     }
 }
 
-macro_rules! expect_netlink_result {
-    ($result:expr, $count:expr) => {
-        if $result.len() != $count {
-            return Err(NetavarkError::msg(format!(
-                "{}: unexpected netlink result (got {} result(s), want {})",
-                function!(),
-                $result.len(),
-                $count
-            )));
-        }
-    };
+pub struct LinkSocket {
+    socket: netlink_sys::Socket,
+    sequence_number: u32,
```

Comment:
one of the reasons I included the buffer in the struct was that rust AFAIK initializes the buffer (`[0; 8192]`) with zeros every time you call it. So I wanted to it only once, of course I have not measured performance just something that I should take a look. I am ok with the change if it is not noticeable.

Reply:
I see your point. 

Just let me know what you prefer and I can try to move ` let mut buffer = [0; 8192];` back into the struct.
Looking at it right now I don't recall if there was a motive behind this change other than the comfort of having a dedicated buffer each time.
But if there was it'll come up during a rework for sure :)